packages/shared_preferences/shared_preferences_tool/lib/src/ui/keys_panel.dart

```diff
@@ -0,0 +1,262 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../async_state.dart';
+import '../shared_preferences_state_provider.dart';
+import 'error_panel.dart';
+
+/// A panel that displays the keys stored in shared preferences.
+class KeysPanel extends StatefulWidget {
+  /// Default constructor for [KeysPanel].
+  const KeysPanel({super.key});
+
+  @override
+  State<KeysPanel> createState() => _KeysPanelState();
+}
+
+class _KeysPanelState extends State<KeysPanel> {
+  bool searching = false;
+  final FocusNode searchFocusNode = FocusNode();
+
+  @override
+  void dispose() {
+    searchFocusNode.dispose();
+    super.dispose();
+  }
+
+  void _startSearching() {
+    setState(() {
+      searching = true;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    void stopSearching() {
+      setState(() {
+        searching = false;
+      });
+      context.sharedPreferencesStateNotifier.filter('');
+    }
+
+    return RoundedOutlinedBorder(
+      clip: true,
+      child: Column(
+        children: <Widget>[
+          AreaPaneHeader(
+            roundedTopBorder: false,
+            includeTopBorder: false,
+            tall: true,
+            title: Row(
+              children: <Widget>[
+                Text(
+                  'Stored Keys',
+                  style: Theme.of(context).textTheme.titleSmall,
+                ),
+                if (searching) ...<Widget>[
+                  const SizedBox(
+                    width: denseSpacing,
+                  ),
+                  Expanded(
+                    child: _SearchField(
+                      searchFocusNode: searchFocusNode,
+                      stopSearching: stopSearching,
+                    ),
+                  ),
+                ] else ...[
+                  const Spacer(),
+                  _ToolbarAction(
+                    tooltipMessage: 'Search',
+                    icon: Icons.search,
+                    onPressed: _startSearching,
+                  ),
+                ],
+                const SizedBox(
+                  width: denseRowSpacing,
+                ),
+                _ToolbarAction(
+                  tooltipMessage: 'Refresh',
+                  icon: Icons.refresh,
+                  onPressed: () {
+                    stopSearching();
+                    context.sharedPreferencesStateNotifier.fetchAllKeys();
+                  },
+                ),
+              ],
+            ),
+          ),
+          const Expanded(
+            child: _StateMapper(),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ToolbarAction extends StatelessWidget {
```

Comment:
@kenzieschmoll I was looking at how the toolbar works for the `Flutter inspector` and it actually uses the [ToolbarAction](https://github.com/flutter/devtools/blob/145f4c2de5647daf09c9d6a3c950e5f00980727a/packages/devtools_app/lib/src/shared/common_widgets.dart#L615) button. But this one is also not exposed in the `devtools_app_shared` package, maybe we should also include it in this [issue](https://github.com/flutter/devtools/issues/7793). This is how it looks like in `Flutter inspector`:
<video src='https://github.com/flutter/packages/assets/11666470/7fe6de5e-4b2b-4f35-a218-76b6995d107f'/>

I can't match exactly the search field without recreating the search component, but I can match the buttons and search field expansion:

|Before|After|
|---|---|
|<video src='https://github.com/flutter/packages/assets/11666470/f6602986-48c1-4f94-a4f2-82b36b8d6a6d'/>|<video src='https://github.com/flutter/packages/assets/11666470/b84ad9f6-7820-407e-a5e1-586000263687'/>|